src/main/java/com/minecolonies/coremod/client/gui/map/ZoomDragMap.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    @Override
+    public void addChild(final Pane child)
+    {
+        super.addChild(child);
+        computeContentSize();
+    }
+
+    public void addChildFirst(final Pane child)
+    {
+        child.setWindow(getWindow());
+        children.add(0, child);
+        adjustChild(child);
+        child.setParentView(this);
+        computeContentSize();
+    }
+
+    /**
+     * Compute the height in pixels of the container.
+     */
+    protected void computeContentSize()
+    {
+        contentHeight = 0;
+        contentWidth = 0;
+
+        for (@NotNull final Pane child : children)
+        {
+            if (child != null)
+            {
+                contentHeight = Math.max(contentHeight, child.getY() + child.getHeight());
+                contentWidth = Math.max(contentWidth, child.getX() + child.getWidth());
+            }
+        }
+
+        // Recompute scroll
+        setScrollY(scrollY);
+        setScrollX(scrollX);
+    }
+
+    @Override
+    public void drawSelf(final PoseStack ms, final double mx, final double my)
+    {
+        scissorsStart(ms, contentWidth, contentHeight);
+
+        ms.pushPose();
+        ms.translate(-scrollX, -scrollY, 0.0d);
+        ms.translate((1 - scale) * x, (1 - scale) * y, 0.0d);
+        final float renderScale = (float) Mth.clamp(scale, 0.26219988382999904, 6d);
+        ms.scale(renderScale, renderScale, 1.0f);
+        super.drawSelf(ms, calcRelativeX(mx), calcRelativeY(my));
+
+        ms.popPose();
+        scissorsEnd(ms);
+    }
+
+    @Override
+    public void drawSelfLast(final PoseStack ms, final double mx, final double my)
+    {
+        scissorsStart(ms, contentWidth, contentHeight);
+
+        ms.pushPose();
+        ms.translate(-scrollX, -scrollY, 0.0d);
+        ms.translate((1 - scale) * x, (1 - scale) * y, 0.0d);
+        final float renderScale = (float) Mth.clamp(scale, 0.26219988382999904, 6d);
+        ms.scale(renderScale, renderScale, 1.0f);
+        super.drawSelfLast(ms, calcRelativeX(mx), calcRelativeY(my));
+        ms.popPose();
+
+        scissorsEnd(ms);
+    }
+
+    private void setScrollY(final double offset)
+    {
+        scrollY = offset;
+    }
+
+    private void setScrollX(final double offset)
+    {
+        scrollX = offset;
+    }
+
+    @Override
+    public boolean onMouseDrag(final double startX, final double startY, final int speed, final double x, final double y)
+    {
+        final boolean childResult = super.onMouseDrag(startX, startY, speed, calcRelativeX(x), calcRelativeY(y));
+        if (!childResult && dragEnabled)
+        {
+            setScrollX(scrollX - x * dragFactor);
+            setScrollY(scrollY - y * dragFactor);
+            return true;
+        }
+        return childResult;
+    }
+
+    @Override
+    public boolean scrollInput(final double wheel, final double mx, final double my)
+    {
+        final boolean childResult = super.scrollInput(wheel, (width - scrollX) / 2, (height - scrollY) / 2);
```

Comment:
![image](https://user-images.githubusercontent.com/17338378/210218968-e4e6af5a-62b7-4d6b-ba27-457b60386299.png)
Now I get why it was acting so weird lmao, can you test with native blockui drag?